mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/util/AnnotationIntrospectionsUtils.java

```diff
@@ -0,0 +1,175 @@
+                                                      getPathModelFromLegacyAnnotation, getPathModelFromSdkAnnotation);
+
+    return ofNullable(pathModel);
+  }
+
+  public static Optional<ClassValueModel> getClassValueModel(Field field) {
+    Function<ClassValue, ClassValueModel> getClassValueFromLegacyAnnotation =
+        (ClassValue classValue) -> new ClassValueModel(Stream.of(classValue.extendsOrImplements()).filter(p -> !isBlank(p))
+            .collect(toList()));
+    Function<org.mule.sdk.api.annotation.param.display.ClassValue, ClassValueModel> getClassValueFromSdkAnnotation =
+        (org.mule.sdk.api.annotation.param.display.ClassValue classValue) -> new ClassValueModel(Stream
+            .of(classValue.extendsOrImplements()).filter(p -> !isBlank(p)).collect(toList()));
+    ClassValueModel classValueModel = getAnnotationValueFromField(
+                                                                  field,
+                                                                  ClassValue.class,
+                                                                  org.mule.sdk.api.annotation.param.display.ClassValue.class,
+                                                                  getClassValueFromLegacyAnnotation,
+                                                                  getClassValueFromSdkAnnotation);
+
+    return ofNullable(classValueModel);
+  }
+
+  public static Optional<Pair<Integer, String>> getPlacementValue(Field field) {
+    Pair<Integer, String> value = getAnnotationValueFromField(field,
+                                                              Placement.class,
+                                                              org.mule.sdk.api.annotation.param.display.Placement.class,
+                                                              (Placement placement) -> Pair.of(placement.order(),
+                                                                                               placement.tab()),
+                                                              (org.mule.sdk.api.annotation.param.display.Placement placement) -> Pair
+                                                                  .of(placement.order(), placement.tab()));
+
+    return ofNullable(value);
+  }
+
+  public static boolean isTextField(Field field) {
+    return isAnnotationPresent(field, Text.class, org.mule.sdk.api.annotation.param.display.Text.class);
+  }
+
+  public static boolean isPasswordField(Field field) {
+    return isAnnotationPresent(field, Password.class);
+  }
+
+  public static boolean isQueryField(Field field) {
+    return isAnnotationPresent(field, Query.class);
+  }
+
+  public static <R extends Annotation> boolean isAnnotationPresent(Field field, Class<R> annotationClass) {
+    return field.getAnnotation(annotationClass) != null;
+  }
+
+  public static <R extends Annotation, S extends Annotation> boolean isAnnotationPresent(Field field,
```

Comment:
should this one be private?

Reply:
Yes, it should. I'm modifying this one since there is already a utils